site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```diff
@@ -3815,9 +3815,10 @@ private void buildLTIToolContextForTemplate(Context context,
 				String ltiToolId = toolMap.get("id").toString();
 				String siteId = StringUtils.trimToNull((String) toolMap.get(m_ltiService.LTI_SITE_ID));
 				toolMap.put("selected", linkedLtiContents.containsKey(ltiToolId));
-				if ( siteId == null || siteId.equals(site.getId()) )
+				if ( siteId == null || (siteId != null && siteId.equals(site.getId())))
```

Comment:
I don't see how this could work, because `||` will shortcut, so it will only test `siteId.equals(site.getId())` if siteId not null. Did you mean to test to see if site is not null?